### DISCUSSION TITLE
Programlama dillerinde random komutu nasıl işliyor?

### DISCUSSION THREAD

**Discussion by BunyaminAkdeniz** [19/classroom#44] [Anonim] [answered]
Kendi rand komutumu yazmaya çalıştığımda uygulama runtime'ı baz alarak bir sayı üreten veya rand sınıfına ait bir değişken tanımlayarak yaptım.
Örneğin;
```ruby
class Rand
  @@count = 0
  def initialize
    @@count += 1
  end

  def get_rand
    #=> @@count to random int 
    ....
  end
end
```
Fakat Ruby'de 
```ruby
prng = Random.new
prng.rand(100) 
```

Bu komut her çalıştığında (runtime aynı olsa bile) başarılı bir şekilde rastgele bir int dönüyor. Merak ettiğim şey bu değeri her seferinde nasıl farklı bir değer alabilir.

**Comment by roktas** [comment, marked as answer]
Tam olarak rastgele değil ama sözde rastgele diyelim: Pseudo Random Number Generators.

Şu video serisini izle: https://www.youtube.com/watch?v=IZ56vqQZux4 (serinin ilki)

**Reply by BunyaminAkdeniz** [reply]
Hocam video serisini izledim. Fakat PRNG algoritmasında da kendi yazdığım Rand algoritmasında ki problemle karşılaşıyorum. Sayıların tamamen rastgele olması için tohum değerinin farklı değerden başlıyor olması gerekiyor. Eğer seed aynı değerle başlarsa tüm sayılar yine aynı olmayacak mı ?

**Reply by roktas** [reply]
Evet aynı olacak, o yüzden "pseudo" zaten.  Aradığın şey "Secure Random".  Tabii bunun bir bedeli var, gerçeğe yakın bir Random üretmek çok zaman alabilir, yeterli "entropi" yoksa.
